### PR TITLE
Add LTensor Hermitian wire length check

### PR DESCRIPTION
**Pull request by multiphaseCFD**
### Before submitting

Please complete the following checklist when submitting a PR:

- [ ] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      [`tests`](../tests) directory!

- [ ] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [ ] Ensure that the test suite passes, by running `make test`.

- [x] Add a new entry to the `.github/CHANGELOG.md` file, summarizing the
      change, and including a link back to the PR.

- [x] Ensure that code is properly formatted by running `make format`. 

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:**

As of cutensornet-v24.03.0, only 1-wire Hermitian observables can be correctly supported. This limit was captured by previous python unit tests. This PR add Hermitian obs wires check in both python and C++ layers and updates the python unit tests to ensure this limit can be captured.

[SC-69138]

**Description of the Change:**

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

File: .github/CHANGELOG.md [hunk 1]
```diff
@@ -37,6 +37,9 @@
 
 ### Bug fixes
 
+* Add check for the number of wires for Hermitian observables in Lightning-Tensor. Only 1-wire Hermitian observables are supported as of cuTensorNet-24.03.0.
+  [(#806)](https://github.com/PennyLaneAI/pennylane-lightning/pull/806)
```
Comment (vincentmr) [inline .github/CHANGELOG.md:41]:
```suggestion
* Check the number of wires for Hermitian observables in Lightning-Tensor. Only 1-wire Hermitian observables are supported as of `cuTensorNet-24.03.0`.
  [(#806)](https://github.com/PennyLaneAI/pennylane-lightning/pull/806)
```

**Review by vincentmr — APPROVED** [review]
LGTM, just a suggestion for the changelog.

File: tests/test_expval.py [hunk 2]
```diff
@@ -163,7 +163,7 @@ def circuit():
         circ_def = qml.QNode(circuit, dev_def)
         assert np.allclose(circ(), circ_def(), tol)
 
-    @pytest.mark.parametrize("n_wires", range(1, 7 if device_name != "lightning.tensor" else 5))
```
Comment (LuisAlfredoNu) [inline tests/test_expval.py:166]: Why do you add this conditional?

**Review by LuisAlfredoNu — APPROVED** [review]
Good work changing the tests

**Pull request merged**